cfme/tests/intelligence/reports/test_canned_corresponds.py

```diff
@@ -88,3 +92,107 @@ def test_cluster_relationships(soft_assert, infra_provider):
+                    vms.power_state == 'on').order_by(vms.name).all()
+
+    assert len(vms_in_db) == len(list(report.data.rows))
+    for vm in vms_in_db:
+        store_path = '{}/{}'.format(vm.storages_name.encode('utf8'),
+                                    vm.vm_location.encode('utf8'))
+        for item in report.data.rows:
+            if vm.vm_name.encode('utf8') == item['VM Name']:
+                assert vm.hosts_name.encode('utf8') == item['Host']
+                assert vm.storages_name.encode('utf8') == item['Datastore']
+                assert store_path == item['Datastore Path']
+                assert (str(vm.vm_last_scan).encode('utf8') == item['Last Analysis Time'] or
+                 (str(vm.vm_last_scan).encode('utf8') == 'None' and
+                 item['Last Analysis Time'] == ''))
+
+
+@pytest.mark.tier(3)
+@test_requirements.report
+@pytest.mark.usefixtures('setup_provider')
+def test_datastores_summary(soft_assert, appliance):
+    """Checks Datastores Summary report with DB data. Checks all data in report, even rounded
+    storage sizes."""
+
+    adb = appliance.db
+    storages = adb['storages']
+    vms = adb['vms']
+    host_storages = adb['host_storages']
+
+    path = ["Configuration Management", "Storage", "Datastores Summary"]
+    report = CannedSavedReport.new(path)
+
+    storages_in_db = adb.session.query(storages.store_type, storages.free_space,
+                                       storages.total_space, storages.name, storages.id).all()
+
+    assert len(storages_in_db) == len(list(report.data.rows))
+    for store in storages_in_db:
+
+        number_of_vms = adb.session.query(vms.id).filter(
+            vms.storage_id == store.id).filter(
+                vms.template == 'f').count()
+        number_of_hosts = adb.session.query(host_storages.host_id).filter(
+            host_storages.storage_id == store.id).count()
+
+        for item in report.data.rows:
+            if store.name.encode('utf8') == item['Datastore Name']:
+                assert store.store_type.encode('utf8') == item['Type']
+                assert round_num(store.free_space) == extract_num(item['Free Space'])
+                assert round_num(store.total_space) == extract_num(item['Total Space'])
+                assert int(number_of_hosts) == int(item['Number of Hosts'])
+                assert int(number_of_vms) == int(item['Number of VMs'])
+
+
+def round_num(column):
+    ret = float(column)
+    by = 1024
+    kb = pow(1024, 2)
+    mb = pow(1024, 3)
+    gb = pow(1024, 4)
+    tb = pow(1024, 5)
+
+    if ret < by:
+        return ret
+
+    if ret >= by and ret < kb:
+        return round((ret / by), 1)
+
+    if ret >= kb and ret < mb:
+        return round((ret / kb), 1)
+
+    if ret >= mb and ret < gb:
+        return round((ret / mb), 1)
+
+    if ret >= gb and ret < tb:
+        return round((ret / gb), 1)
+
+    if ret >= tb:
+        return round((ret / tb), 1)
```

Comment:
Make it if/elif/elif/elif/else with an exception raise in else so in case of a missing condition it gets visible. Btw, do you think these functions could live somewhere outside of tests?